### PR TITLE
cap1xxx: generalize cap1188 to support more devices

### PR DESCRIPTION
**Pull request by maruel**
cap1166 can be found in Display-O-Tron HAT by Pimoroni.

### CODE REVIEW

**Comment by maruel** [comment]
@jdortiz this should unblock you.

File: experimental/devices/cap1xxx/cap1xxx.go [hunk 1]
```diff
@@ -194,12 +217,13 @@ func (d *Dev) SetLED(idx int, state bool) error {
 	if d.opts.LinkedLEDs {
 		return wrapf("can't manually set LEDs when they are linked to sensors")
 	}
-	if idx > 7 || idx < 0 {
+	if idx > d.numLEDs || idx < 0 {
```
Comment (mattetti) [inline experimental/devices/cap1xxx/cap1xxx.go]: I believe this should be `>=` or `d.numLEDs - 1`

File: experimental/devices/cap1xxx/cap1xxx.go [hunk 2]
```diff
@@ -259,10 +283,62 @@ func makeDev(c conn.Conn, isSPI bool, opts *Opts) (*Dev, error) {
...
 
 	// Read the product id to confirm it matches our expectations.
-	if productID, err := d.c.ReadUint8(0xFD); err != nil {
+	productID, err := d.c.ReadUint8(0xFD)
+	if err != nil {
 		return nil, wrapf("failed to read product id: %v", err)
-	} else if productID != 0x50 {
-		return nil, wrapf("unexpected chip id %x; is this a cap1188?", productID)
+	}
+	switch productID {
```
Comment (mattetti) [inline experimental/devices/cap1xxx/cap1xxx.go]: 👍

**Comment by jdortiz** [comment]
Cool, thanks!

**Pull request merged**